modules/persistenceService/src/main/scala/subsPersistenceLayer/SubscriptionServicePersistenceLayer.scala

```diff
@@ -0,0 +1,52 @@
+package subsPersistenceLayer
+
+import doobie.{ConnectionIO, Query0}
+import doobie.implicits._
+import subscriptionAlgebra.SubscriptionServiceAlgebra
+import utils.Types.{GetSubscriptionData, Id, PostSubscriptionData, PostSubscriptions, RepositoryId}
+import cats.implicits._
+import doobie.util.update.Update
+import doobie.postgres.implicits._
+
+object SubscriptionServicePersistenceLayer {
+  val subscriptionServiceAlgImp: SubscriptionServiceAlgebra[ConnectionIO] =
+    new SubscriptionServiceAlgebra[ConnectionIO] {
+      def get(id: Id): ConnectionIO[List[GetSubscriptionData]] =
+        SubscriptionServiceQuery.get(id).to[List]
+
+      def post(id: Id, subscriptions: PostSubscriptions): ConnectionIO[Unit] =
+        for {
+          subs <- subscriptions.subscriptions.traverse { postData =>
+            SubscriptionServiceQuery
+              .post(postData)
+              .withUniqueGeneratedKeys[RepositoryId]("repository_id")
+              .map(repoInfo => (id, repoInfo))
+          }
+          _ <- SubscriptionServiceQuery.insertInSubscription.updateMany(subs).void
+        } yield ()
+
+      def delete(id: Id, subscriptions: PostSubscriptions): ConnectionIO[Unit] =
+        SubscriptionServiceQuery.delete(id).updateMany(subscriptions.subscriptions).void
+    }
+}
+
+object SubscriptionServiceQuery {
+  def get(id: Id): Query0[GetSubscriptionData] =
+    sql"select owner, repository, subscribed_at::varchar from repositories join subscriptions using (repository_id) where user_id = ${id.id}"
```

Comment:
To be honest I'm not a big fan of the type mapping here of `subscribed_at`. What kind of error are you getting?

Reply:
Thanks @fedefernandez 
Please this is because my `case class` expects `subscribed_at` to be of type `String`.
`ADD COLUMN subscribed_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP;`

```
[info]           TIMESTAMP (timestamp) is ostensibly coercible to String according
[info]           to the JDBC specification but is not a recommended target type.
[info]           Expected schema type was CHAR or VARCHAR or LONGVARCHAR or NCHAR
[info]           or NVARCHAR or LONGNVARCHAR. (Checker.scala:56)
```

<img width="1146" alt="Screenshot 2021-08-17 at 18 45 54" src="https://user-images.githubusercontent.com/37377831/129775466-dee1505d-f18d-499b-a716-034a0f78e9fb.png">

Reply:
How about this:

```diff
+ import java.time.OffsetDateTime

- final case class SubscribeAt(subscribeAt: String) extends AnyVal
+ final case class SubscribeAt(value: OffsetDateTime) extends AnyVal

  object SubscribeAt {
    implicit val decoder: Decoder[SubscribeAt] = deriveUnwrappedDecoder[SubscribeAt]
    implicit val encoder: Encoder[SubscribeAt] = deriveUnwrappedEncoder[SubscribeAt]
  }
```

This way you don't need a type cast, plus you work with time objects in your code, better than raw strings.

Reply:
Thank you very much @moleike